container_linux_test.go

```diff
@@ -871,70 +875,6 @@ func TestContainerKillAll(t *testing.T) {
 	}
 }
 
-func TestShimSigkilled(t *testing.T) {
```

Comment:
I looked at the test and it really wasn't testing much, killing the shim then killing the process.  I don't think it was a valid test or it was and changed sometime in the past.  Felt safe to remove.

Reply:
It checks that the container process (redis) is correctly cleaned up if the shim crash/is sigkilled (e.g. by oom).
It doesn't actually kill the process, it checks that it died.

Reply:
ok, my bad, i'll fix it.  i missed the kill 0

Reply:
Seems this test is still removed in the latest commit? or did I miss a move somewhere else?

Reply:
ok, so I researched this more and the test is only valid for v1 when container was still running and didn't have to reconnect to a shim again.  The reason is that it sets an exit handler that does the "cleanup on dead shim" code.  However, this only runs if the shim process that containerd is doing a waidpid on returns and checks the shim pid.  

I think its safe to remove this as it doesn't really work, it only works in the tests because the daemon isn't restarted.  If we want to keep it, i can do some refactoring but it will take a little bit of work to make this work across daemon reboots with running shims.  I can do that in a followup PR.

Reply:
Makes sense.

I haven't yet got the time to compile and play around with the new API to see how it actually behave.

Having a way to start a "cleanup shim" in such scenario may be a cleaner solution for v2 though (and if that fails, things are left to the user to clean).